src/tdigest.rs

```diff
@@ -153,9 +153,9 @@ pub fn tdigest_deserialize(
     tdigest.into()
 }
 
-flat_serialize_macro::flat_serialize! {
-    struct TsTDigestData {
-        header: u32,
+crate::pg_type! {
+    #[derive(Debug)]
+    struct TimescaleTDigest: TsTDigestData {
```

Comment:
I really like everything about this, except for this `: TsTDigestData`.  If there was a way to completely hide the inner type (probably requiring another macro in the final function) this would be perfect.